.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,29 @@
+Please start by choosing an appropriate base branch for your pull request.
+
+Making a quick fix for a problem observed on production service? Fixing a minor
+bug, improving existing documentation or record metadata? Make your pull request
+against the ``production`` branch.
+
+Introducing a new feature? Make your pull request against the ``master`` branch.
+
+Please reference all issues that this PR addresses or closes. Do not forget to
+include ``(addresses #123)`` or ``(closes #123)`` in your commit log message.
+
+Please run the tests locally in order to speed up developments and save Travis
+CI cycles.  Look for ``./run-tests.sh``.
+
+Finally, before submitting the pull request, please select an appropriate issue
+labels and the concrete milestone to ease our housekeeping.
+
+Checklist:
+- [ ] Verified that the system builds and works locally as described?
+- [ ] Added tests to ensure success and failure?
+- [ ] Updated the documentation?
+- [ ] Described the nature of the change in a self-understandable commit log
+      message?
+- [ ] Provided all information to facilitate review and ensure fast issue flow?
```

Comment:
This last point is a bit vague.

Reply:
Also, what happens if I submit a PR for some minor changes? I won't need to check these, right? Do we expect many non-CERN developers contributing? Maybe we can remove this checklist or explicitly state it's for developers? I don't expect documentation or metadata changes to need all these items.

Reply:
The checklist is mostly for developers.  It does not hurt to home in the message which things should be usually done before we make PRs 😄  We'll see whether we retain the checklist "as is" with each PR, so that the author will tick items as appropriate for the reviewer to see about the status, or whether this is only for the initial information of the authorr who would be free to delete the whole blob.